plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts

```diff
@@ -31,6 +31,40 @@ export function createRouter({
     response.send(await argoSvc.findArgoApp({ name: argoAppName }));
   });
 
+  router.get(
+    '/argoInstance/:argoInstanceName/applications/name/:argoAppName/revisions/:revisionID/metadata',
+    async (request, response) => {
+      const revisionID: string = request.params.revisionID;
+      const argoInstanceName: string = request.params.argoInstanceName;
+      const argoAppName = request.params.argoAppName;
+      logger.info(`Getting info on ${argoAppName}`);
+      logger.info(`Getting app ${argoAppName} on ${argoInstanceName}`);
+      const argoInstanceArray = argoSvc.getArgoInstanceArray();
+      const matchedArgoInstance = argoInstanceArray.find(
+        argoInstance => argoInstance.name === argoInstanceName,
+      );
+      if (matchedArgoInstance === undefined) {
+        return response.status(500).send({
+          status: 'failed',
+          message: 'cannot find an argo instance to match this cluster',
+        });
+      }
+      let token: string;
+      if (!matchedArgoInstance.token) {
+        token = await argoSvc.getArgoToken(matchedArgoInstance);
+      } else {
+        token = matchedArgoInstance.token;
+      }
```

Comment:
`token` has already been declared as `string` type, right?

Reply:
sorry for the misunderstanding. I was not clear enough. please read (you have seen this comment before) 'I see this code is used in several places, instead of copying and pasting to reuse it, may you please refactor? I am a firm believer in the Boy Scouts Rule - Leave the campsite better than how it was found - Always leave the code better than you found it."